### PR TITLE
webview: allow setting background color

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes https://github.com/atom/electron/issues/1397

Based on this chromium [issue](https://bugs.chromium.org/p/chromium/issues/detail?id=589509#c4) , setting non opaque values to `content::RenderWidgetHostView::SetBackgroundColor` results in default UA dependent background color. This patch overrides the behavior, so setting background color to webview tag will now be respected. Also have removed the undocumented `allowtransparency` attribute. 

**Note:** Default background color when not specified for webview will now be transparent, instead of white in Electron, with this patch.

### CODE REVIEW

File: atom/renderer/atom_render_view_observer.cc [hunk 1]
```diff
@@ -148,4 +150,10 @@ void AtomRenderViewObserver::OnBrowserMessage(const base::string16& channel,
   }
 }
 
+void AtomRenderViewObserver::OnSetTransparentBackground() {
+  if (!render_view()->GetWebView())
+    return;
+  render_view()->GetWebView()->setBaseBackgroundColor(0x00000000);
```
Comment (zcbenz) [inline atom/renderer/atom_render_view_observer.cc]: We can use `SK_ColorTRANSPARENT`.

Comment (deepak1556) [inline atom/renderer/atom_render_view_observer.cc]: Fixed, thanks!

File: atom/browser/web_view_guest_delegate.cc [hunk 2]
```diff
@@ -121,16 +104,9 @@ void WebViewGuestDelegate::HandleKeyboardEvent(
...
-  // interstitial page if one is showing at this time. We only want opacity
-  // to apply to web pages.
-  auto render_view_host_view =
-      web_contents()->GetRenderViewHost()->GetWidget()->GetView();
-  if (guest_opaque_)
-    render_view_host_view->SetBackgroundColorToDefault();
-  else
-    render_view_host_view->SetBackgroundColor(SK_ColorTRANSPARENT);
+  // Set default UA-dependent background as transparent.
+  api_web_contents_->Send(new AtomViewMsg_SetTransparentBackground(
```
Comment (zcbenz) [inline atom/browser/web_view_guest_delegate.cc]: I think we can just call this in `AtomRendererClient::RenderViewCreated` instead of sending a message.

Comment (deepak1556) [inline atom/browser/web_view_guest_delegate.cc]: Wasn't sure if we wanted to acheive this behavior for all render views. Fixed.

**Comment by zcbenz** [comment]
:+1:

**Comment by philipgiuliani** [comment]
Would it also be possible to set a background image? For a good looking splash screen?

**Comment by pavankumarkatakam** [comment]
yeah, splash screen idea is nice.

**Pull request merged**